consensus/src/dag/types.rs

```diff
@@ -197,8 +239,15 @@ impl Node {
 
 impl TDAGMessage for Node {
     fn verify(&self, verifier: &ValidatorVerifier) -> anyhow::Result<()> {
+        ensure!(self.digest() == self.calculate_digest(), "invalid digest");
```

Comment:
Does it make sense to move this check after we validate it?
To prevent DDoS. If I already signed a node for you in this round I do not need to verify and compute expensive digest.